tempesta_fw/sock_srv.c

```diff
@@ -119,13 +119,13 @@ tfw_sock_srv_connect_try(TfwSrvConnection *srv_conn)
 	}
 
 	sock_set_flag(sk, SOCK_DBG);
-	tfw_connection_link_sk(&srv_conn->conn, sk);
+	tfw_connection_link_from_sk(conn, sk);
 	ss_set_callbacks(sk);
 
 	r = ss_connect(sk, &addr->sa, tfw_addr_sa_len(addr), 0);
 	if (r) {
 		TFW_ERR("can't initiate a connect to server: error %d\n", r);
-		tfw_connection_unlink_sk(&srv_conn->conn);
+		tfw_connection_unlink_sk(conn, sk);
```

Comment:
You can add assertion ```conn->sk == sk``` just before call of the function or use define wrapper like tfw_connection_unlink_sk_check().